storage/index.ts

```diff
@@ -17,36 +17,62 @@ type UploadResult = import('firebase/storage').UploadResult;
 export function fromTask(
     task: UploadTask,
 ): Observable<UploadTaskSnapshot> {
-  return new Observable<UploadTaskSnapshot>((subscriber) => {
-    const progress = (snap: UploadTaskSnapshot): void => subscriber.next(snap);
-    const error = (e: Error): void => subscriber.error(e);
-    const complete = (): void => subscriber.complete();
-    // emit the current state of the task
-    progress(task.snapshot);
-    // emit progression of the task
-    const unsubscribeFromOnStateChanged = task.on('state_changed', progress);
-    // use the promise form of task, to get the last success snapshot
-    task.then(
-        (snapshot) => {
-          progress(snapshot);
-          setTimeout(() => complete(), 0);
-        },
-        (e) => {
-          progress(task.snapshot);
-          setTimeout(() => error(e), 0);
-        },
-    );
-    // the unsubscribe method returns by storage isn't typed in the
-    // way rxjs expects, Function vs () => void, so wrap it
-    return function unsubscribe() {
-      unsubscribeFromOnStateChanged();
+  return new Observable<UploadTaskSnapshot>(subscriber => {
+    let lastSnapshot: UploadTaskSnapshot | null = null;
+    let complete = false;
+    let hasError = false;
+    let error: any = null;
+
+    const emit = (snapshot: UploadTaskSnapshot) => {
+      lastSnapshot = snapshot;
+      schedule();
     };
-  }).pipe(
-      // since we're emitting first the current snapshot and then progression
-      // it's possible that we could double fire synchronously; namely when in
-      // a terminal state (success, error, canceled). Debounce to address.
-      debounceTime(0),
-  );
+
+    let id: ReturnType<typeof setTimeout> | null = null;
+
+    /**
+     * Schedules an async event to check and emit
+     * the most recent snapshot, and complete or error 
+     * if necessary.
+     */
+    const schedule = () => {
+      if (!id) {
+        id = setTimeout(() => {
+          id = null;
+          if (lastSnapshot) subscriber.next(lastSnapshot);
+          if (complete) subscriber.complete();
+          if (hasError) subscriber.error(error);
+        });
+      }
+    };
+
+    subscriber.add(() => {
+      // If we have any emissions checks scheduled, cancel them.
+      if (id) clearTimeout(id);
+    });
+
+    // Emit the initial snapshot
+    emit(task.snapshot);
+
+    // Take each update and schedule them to be emitted (see `emit`)
+    subscriber.add(task.on('state_changed', emit) as () => void);
+
+    // task is a promise, so we can convert that to an observable,
+    // this is done for the ergonomics around making sure we don't
+    // try to push errors or completions through closed subscribers.
```

Comment:
```suggestion
    // try to push errors or completions through closed subscribers
```

Just trying to make a blank commit to fix CI.